### PR TITLE
Add ability to benchmark pull requests

### PR DESCRIPTION
**Pull request by rashidnhm**
**Context:** Add ability to benchmark a pull request

**Description of the Change:**
This pull request adds the ability to benchmark a PR.

When a user with either `admin` or `maintain` permissions comments `/benchmark` on a pull request, this workflow will trigger a webhook to benchmarks repo with sufficient details to benchmark the PR, it is then up to the benchmarks action to report back with results.

Once the webhook is sent, a :+1: reaction is added to the comment to indicate it was successful.

**Benefits:**
New feature.

**Possible Drawbacks:**
This is an MVP product, we may want to iterate on this to add further information or functionality. However, the current state is sufficient for the basic needs of benchmarking.

**Related GitHub Issues:**
None. [sc-73314](https://app.shortcut.com/xanaduai/story/73314/add-webhook-capability-to-pennylane-prs-to-trigger-benchmark-build)

### CODE REVIEW

File: .github/workflows/trigger_benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        id: comment_body
+        run: echo "triggers_benchmark=${{ github.event.comment.body == env.BENCHMARK_TRIGGER_COMMAND }}" >> $GITHUB_OUTPUT
+
+      - name: Set Repository Info
+        id: repo_info
+        env:
+          REPO_OWNER: ${{ github.repository_owner }}
+        run: |
+          echo "REPO_OWNER=$REPO_OWNER" >> $GITHUB_OUTPUT
+          echo "REPO_NAME=${GITHUB_REPOSITORY#$REPO_OWNER/}" >> $GITHUB_OUTPUT
```
Comment (mudit2812) [inline .github/workflows/trigger_benchmarks.yml:28]: `GITHUB_REPOSITORY` hasn't been set as an env var.

Comment (rashidnhm) [reply .github/workflows/trigger_benchmarks.yml:28]: Hey Mudit, `GITHUB_REPOSITORY` is set by GitHub Actions itself and is available to all shell scripts within actions (it is the equivalent to doing `${{ github.environment }}`

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables
image

Comment (mudit2812) [reply .github/workflows/trigger_benchmarks.yml:28]: Nice! Thanks for the info 😄

**Review by jay-selby — APPROVED** [review]
LGTM.  Thanks Rashid!

I think there is some future work on the benchmark side to cleanly format the output reported back to the PR.

**Comment by rashidnhm** [comment]
> Looks good to me, but is it possible to try it out before we merge? Just want to see it in action to verify it works. The code itself looks fine to me, but I also don't use javascript much. Or is it not possible since the PR on the benchmarks suite-side is still open?

I have tried to get the code to work as much as I could in a private fork ... however, yes, none of the changes here can be tested as part PR unfortunately

**Comment by mudit2812** [comment]
> > Looks good to me, but is it possible to try it out before we merge? Just want to see it in action to verify it works. The code itself looks fine to me, but I also don't use javascript much. Or is it not possible since the PR on the benchmarks suite-side is still open?
> 
> I have tried to get the code to work as much as I could in a private fork ... however, yes, none of the changes here can be tested as part PR unfortunately

All good then. This is all pretty early iteration work anyway, we can always come back later with bug fixes 😄

**Pull request merged**